fuel-p2p/src/orchestrator.rs

```diff
@@ -0,0 +1,98 @@
+use fuel_core_interfaces::model::Vote;
+use fuel_tx::Transaction;
+use libp2p::identity::Keypair;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+use crate::{
+    behavior::FuelBehaviourEvent,
+    config::P2PConfig,
+    gossipsub::messages::GossipsubMessage,
+    service::{FuelP2PEvent, FuelP2PService},
+};
+
+pub struct NetworkOrchestrator {
+    p2p_service: FuelP2PService,
+
+    // receivers
+    rx_block_producer: Receiver<()>,
+    rx_bft: Receiver<Vote>,
+
+    // senders
+    tx_synchronizer: Sender<()>,
+    tx_transaction_pool: Sender<Transaction>,
+    tx_bft: Sender<()>,
+}
+
+impl NetworkOrchestrator {
+    pub async fn new(
+        local_keypair: Keypair,
+        p2p_config: P2PConfig,
+        rx_block_producer: Receiver<()>,
+        rx_bft: Receiver<Vote>,
+
+        tx_synchronizer: Sender<()>,
+        tx_transaction_pool: Sender<Transaction>,
+        tx_bft: Sender<()>,
+    ) -> Self {
+        let p2p_service = FuelP2PService::new(local_keypair, p2p_config)
+            .await
+            .unwrap();
+
+        Self {
+            p2p_service,
+
+            rx_bft,
+            rx_block_producer,
+
+            tx_bft,
+            tx_synchronizer,
+            tx_transaction_pool,
+        }
+    }
+
+    pub async fn run(&mut self) {
+        loop {
+            tokio::select! {
+                p2p_event = self.p2p_service.next_event() => {
```

Comment:
instead of nesting all these match patterns, maybe create a single level identity who will check the cases?